### PR TITLE
feature(portfolio): Image invalidate

### PR DESCRIPTION
**Pull request by jorbuedo**
## Description / Change(s) / Related issue(s)
Adds back option to invalidate image by swipe down

##### Ticket

[YOMO-1859](https://emurgo.atlassian.net/browse/YOMO-1859)

[YOMO-1859]: https://emurgo.atlassian.net/browse/YOMO-1859?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: apps/wallet-mobile/src/features/Portfolio/common/hooks/usePortfolioImage.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+export const usePortfolioImageInvalidate = () => {
+  const {
+    networkManager: {tokenManager},
+  } = useSelectedNetwork()
+  const mutation = useMutation({
+    mutationFn: async (ids: Array<Portfolio.Token.Id>) => {
+      const errors = await tokenManager.api.tokenImageInvalidate(ids)
+      await Image.clearDiskCache()
+      await Image.clearMemoryCache()
+      console.log(errors)
```
Comment (stackchain) [inline apps/wallet-mobile/src/features/Portfolio/common/hooks/usePortfolioImage.ts]:
```suggestion
```

File: apps/wallet-mobile/src/features/Nfts/useCases/Nfts.tsx [hunk 2]
```diff
@@ -40,6 +41,12 @@ export const Nfts = () => {
     return isSearching ? balances.nfts.filter(({info}) => byName(info)) : balances.nfts
   }, [balances.nfts, isSearching, search])
 
+  const {invalidate, isLoading} = usePortfolioImageInvalidate()
+  const onRefresh = React.useMemo(
```
Comment (stackchain) [inline apps/wallet-mobile/src/features/Nfts/useCases/Nfts.tsx]: it'd invalidate all list, for now I'd say to add it to the `NFTDetails/MediaDetails` only, (I'm afraid that if the users starting pulling it too often we will end up reprocessing tons of images which a little bit expensive).

Please add it to the TokenList (details) inside the Portfolio too, so over there we can also invalidate the tokens (I dunno if it is using the MediaDetails) if it's just disregard.

File: packages/portfolio/src/adapters/dullahan-api/api-maker.ts [hunk 3]
```diff
@@ -336,6 +366,8 @@ export const apiConfig: ApiConfig = freeze(
         'https://yoroi-backend-zero-sanchonet.emurgornd.com/tokens/activity/multi',
       tokenPriceHistory:
         'https://add50d9d-76d7-47b7-b17f-e34021f63a02.mock.pstmn.io/v1/token-price-history',
+      tokenImageInvalidate:
+        'https://preprod.processed-media.yoroiwallet.com/invalidate',
```
Comment (stackchain) [inline packages/portfolio/src/adapters/dullahan-api/api-maker.ts:362]: sancho is about to die, anyways.
```suggestion
        'https://sancho.processed-media.yoroiwallet.com/invalidate',
```

File: apps/wallet-mobile/src/features/Nfts/useCases/Nfts.tsx (hunk 2 above)
Comment (jorbuedo) [reply apps/wallet-mobile/src/features/Nfts/useCases/Nfts.tsx]: Same component.
I'll remove the mass invalidate from the gallery.

File: packages/portfolio/src/adapters/dullahan-api/api-maker.ts (hunk 3 above)
Comment (jorbuedo) [reply packages/portfolio/src/adapters/dullahan-api/api-maker.ts:362]: That api doesn't have a `sancho` endpoint

**Review by stackchain — APPROVED** [review]

**Pull request merged**